### PR TITLE
Update GL_GetSwapInterval for backends

### PR DESCRIPTION
**Pull request by 1bsyl**
update the backends

following https://github.com/libsdl-org/SDL/pull/6939
https://github.com/libsdl-org/SDL/issues/2158

not really tested ...

### CODE REVIEW

**Comment by 1bsyl** [comment]
glXQueryDrawable() would call some exit() upon error

File: src/video/cocoa/SDL_cocoaopengl.m [hunk 1]
```diff
@@ -467,11 +467,16 @@ int Cocoa_GL_SetSwapInterval(_THIS, int interval)
...
+int Cocoa_GL_GetSwapInterval(_THIS, int *interval)
 {
     @autoreleasepool {
         SDLOpenGLContext *nscontext = (__bridge SDLOpenGLContext *)SDL_GL_GetCurrentContext();
-        return nscontext ? SDL_AtomicGet(&nscontext->swapIntervalSetting) : 0;
+        if (nscontext) {
+            *interval = SDL_AtomicGet(&nscontext->swapIntervalSetting);
+            return 0;
+        } else {
+            return -1;
```
Comment (slouken) [inline src/video/cocoa/SDL_cocoaopengl.m:478]: This should set an error

Comment (1bsyl) [reply src/video/cocoa/SDL_cocoaopengl.m:479]: done

Comment (slouken) [reply src/video/cocoa/SDL_cocoaopengl.m:479]: How about:
`return SDL_SetError("no OpenGL context");`

Comment (1bsyl) [reply src/video/cocoa/SDL_cocoaopengl.m]: done

**Pull request merged**